service/src/main/java/uk/gov/hmcts/reform/fpl/interceptors/CafcassApiInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.fpl.interceptors;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.hmcts.reform.fpl.exceptions.api.AuthorizationException;
+import uk.gov.hmcts.reform.fpl.exceptions.api.ServiceUnavailableException;
+import uk.gov.hmcts.reform.fpl.model.cafcass.api.CafcassApiFeatureFlag;
+import uk.gov.hmcts.reform.fpl.service.FeatureToggleService;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import java.util.Objects;
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.UserRole.CAFCASS_SYSTEM_UPDATE;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CafcassApiInterceptor implements HandlerInterceptor {
+    private final FeatureToggleService featureToggleService;
+    private final ObjectProvider<IdamClient> idamClient;
+
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+                             Object handler) throws Exception {
+        if (!isCafcassApiToggledOn()) {
+            log.info("Cafcass API is disabled.");
+            throw new ServiceUnavailableException();
+        }
+
+        String authToken = request.getHeader("Authorization");
+        if (isNotEmpty(authToken)) {
+            UserInfo userInfo = Objects.requireNonNull(idamClient.getIfAvailable()).getUserInfo(authToken);
```

Comment:
we may need to cache this before proper release - when they do lots of requests for documents one after another this will result in lots of calls to IDAM unnecessarily.